posthog-core/src/index.ts

```diff
@@ -550,7 +550,11 @@ export abstract class PostHogCoreStateless {
           // fetch will only throw on network errors or on timeouts
           throw new PostHogFetchNetworkError(e)
         }
-        if (res.status < 200 || res.status >= 400) {
+        // If we're in no-cors mode, we can't access the response status
+        // We only throw on HTTP errors if we're not in no-cors mode
+        // https://developer.mozilla.org/en-US/docs/Web/API/Request/mode#no-cors
+        const isNoCors = options.mode === 'no-cors'
+        if (!isNoCors && (res.status < 200 || res.status >= 400)) {
```

Comment:
important bits are here, we were checking the status although it was always 0, we were retrying 3 times every time, events were not duplicated because it was the very same id.